samples/nrf_desktop/src/modules/hid_state.c

```diff
@@ -391,52 +391,49 @@ static void connect_subscriber(const void *subscriber_id, bool is_usb)
 
 static void disconnect_subscriber(const void *subscriber_id)
```

Comment:
Changes in this function are not related to the bug but I did a small rework to make it more readable. There is a one change in logic that should not change how program behaves. I will point it out below.